### PR TITLE
feat: log_attributes @ event level

### PR DESCRIPTION
**Pull request by xCodeMuse**
Changes made:

- Added ability to add custom logs @ event level.
- can override default logs

[pending]: WORK_IN_PROGRESS
 custom error_type log_attributes

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -292,13 +293,25 @@ async function main() {
...
+      let obj = eval(`events[event.type]?.log_attributes?.${key}`);
+      let filter = `${key}`
+      if(obj){
+        // do nothing
+      }else{
+        obj = eval(`event.data`);
+        filter = `${logAttributes[key]}`
+      }
+      // JQ.run
+      await run(`.${filter}`,JSON.stringify(obj),{input:'string',output:'json'}).then((output) =>{
```
Comment (sambamindgrep) [inline src/app.ts:311]: lets not pass input as string, lets pass obj as it is and say input as json and check

File: src/app.ts [hunk 2]
```diff
@@ -292,13 +293,25 @@ async function main() {
...
     const logAttributes = (config as any).log_attributes || [];
 
+
     for (const key in logAttributes) {
-      const obj = `event.data?.${logAttributes[key]}`;
-      // eslint-disable-next-line no-eval
-      childLogAttributes[key] = eval(obj);
+      let obj = eval(`events[event.type]?.log_attributes?.${key}`);
+      let filter = `${key}`
+      if(obj){
```
Comment (sambamindgrep) [inline src/app.ts:304]: use spaces

File: src/app.ts [hunk 3]
```diff
@@ -292,13 +293,25 @@ async function main() {
...
+    childLogAttributes.file_name = events[event.type].fn
+    
     const logAttributes = (config as any).log_attributes || [];
 
+
     for (const key in logAttributes) {
-      const obj = `event.data?.${logAttributes[key]}`;
-      // eslint-disable-next-line no-eval
-      childLogAttributes[key] = eval(obj);
+      let obj = eval(`events[event.type]?.log_attributes?.${key}`);
```
Comment (sambamindgrep) [inline src/app.ts:302]: obj is just event.data

Comment (xCodeMuse) [reply src/app.ts:302]: i did it to get the event level logs to override default config. am i doing it wrong?

File: src/app.ts [hunk 4]
```diff
@@ -292,13 +293,26 @@ async function main() {
...
+      let filter = `${key}`
+
+      if(obj) {
+        // do nothing
+      } else {
+        obj = eval(`event.data`);
+        filter = `${logAttributes[key]}`
+      }
+      // Jsonata
+      const extractFilteredData = jsonata(filter);
```
Comment (sambamindgrep) [inline src/app.ts:313]: Compile this and store and use this to evaluate rather than compiling for every log line

**Review by mastersilv3r — APPROVED** [review]

**Pull request merged**